src/m365/pp/commands/tenant/tenant-settings-set.ts

```diff
@@ -0,0 +1,331 @@
+      },
+      {
+        option: '--enableModelDataSharing [enableModelDataSharing]',
+        autocomplete: ['true', 'false']
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.boolean.push('walkMeOptOut', 'disableNPSCommentsReachout', 'disableNewsletterSendout', 'disableEnvironmentCreationByNonAdminUsers', 'disablePortalsCreationByNonAdminUsers', 'disableSurveyFeedback', 'disableTrialEnvironmentCreationByNonAdminUsers', 'disableCapacityAllocationByEnvironmentAdmins', 'disableSupportTicketsVisibleByAllUsers', 'disableDocsSearch', 'disableCommunitySearch', 'disableBingVideoSearch', 'disableShareWithEveryone', 'enableGuestsToMake', 'disableMembersIndicator', 'disableMakerMatch', 'disablePreferredDataLocationForTeamsEnvironment', 'disableAdminDigest', 'disableDeveloperEnvironmentCreationByNonAdminUsers', 'disableBillingPolicyCreationByNonAdminUsers', 'disableChampionsInvitationReachout', 'disableSkillsMatchInvitationReachout', 'disableCopilot', 'enableOpenAiBotPublishing', 'enableModelDataSharing');
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+
+        if (args.options.shareWithColleaguesUserLimit) {
+          const shareWithColleaguesUserLimit: number = parseInt(args.options.shareWithColleaguesUserLimit);
+          if (isNaN(shareWithColleaguesUserLimit) || shareWithColleaguesUserLimit < 0) {
+            return `'${args.options.shareWithColleaguesUserLimit}' is not a valid number`;
+          }
+        }
+
+        if (args.options.storageCapacityConsumptionWarningThreshold) {
+          const storageCapacityConsumptionWarningThreshold: number = parseInt(args.options.storageCapacityConsumptionWarningThreshold);
+          if (isNaN(storageCapacityConsumptionWarningThreshold) || storageCapacityConsumptionWarningThreshold < 0) {
+            return `'${args.options.storageCapacityConsumptionWarningThreshold}' is not a valid number`;
+          }
+        }
+
+        if (typeof args.options.walkMeOptOut === 'undefined' &&
+          typeof args.options.disableNPSCommentsReachout === 'undefined' &&
+          typeof args.options.disableNewsletterSendout === 'undefined' &&
+          typeof args.options.disableEnvironmentCreationByNonAdminUsers === 'undefined' &&
+          typeof args.options.disablePortalsCreationByNonAdminUsers === 'undefined' &&
+          typeof args.options.disableSurveyFeedback === 'undefined' &&
+          typeof args.options.disableTrialEnvironmentCreationByNonAdminUsers === 'undefined' &&
+          typeof args.options.disableCapacityAllocationByEnvironmentAdmins === 'undefined' &&
+          typeof args.options.disableSupportTicketsVisibleByAllUsers === 'undefined' &&
+          typeof args.options.disableDocsSearch === 'undefined' &&
+          typeof args.options.disableCommunitySearch === 'undefined' &&
+          typeof args.options.disableBingVideoSearch === 'undefined' &&
+          !args.options.shareWithColleaguesUserLimit &&
```

Comment:
Currently providing the value `0` will fail this validation while it's technically a valid value.